core/src/main/java/io/questdb/cutlass/line/tcp/LineTcpMeasurementScheduler.java

```diff
@@ -0,0 +1,812 @@
+            pubSeq = null;
+            tableUpdateDetailsByTableName.clear();
+            for (int n = 0; n < queue.getCapacity(); n++) {
+                queue.get(n).close();
+            }
+        }
+    }
+
+    static class LineTcpMeasurementEvent implements Closeable {
+        private TruncatedLineProtoLexer lexer;
+        private final CharSequenceCache cache;
+        private final MicrosecondClock clock;
+        private final LineProtoTimestampAdapter timestampAdapter;
+        private long measurementNameAddress;
+        private final LongList addresses = new LongList();
+        private int firstFieldIndex;
+        private long timestampAddress;
+        private int errorPosition;
+        private int errorCode;
+        private int threadId;
+        private long timestamp;
+
+        private int rebalanceFromThreadId;
+        private int rebalanceToThreadId;
+        private String rebalanceTableName;
+        private boolean rebalanceReleasedByFromThread;
+
+        private LineTcpMeasurementEvent(int maxMeasurementSize, MicrosecondClock clock, LineProtoTimestampAdapter timestampAdapter) {
+            lexer = new TruncatedLineProtoLexer(maxMeasurementSize);
+            cache = lexer.getCharSequenceCache();
+            this.clock = clock;
+            this.timestampAdapter = timestampAdapter;
+            lexer.withParser(new LineProtoParser() {
+                @Override
+                public void onLineEnd(CharSequenceCache cache) {
+                }
+
+                @Override
+                public void onEvent(CachedCharSequence token, int type, CharSequenceCache cache) {
+                    assert cache == LineTcpMeasurementEvent.this.cache;
+                    switch (type) {
+                        case EVT_MEASUREMENT:
+                            assert measurementNameAddress == 0;
+                            measurementNameAddress = token.getCacheAddress();
+                            break;
+
+                        case EVT_TAG_NAME:
+                        case EVT_TAG_VALUE:
+                            assert firstFieldIndex == -1;
+                            addresses.add(token.getCacheAddress());
+                            break;
+                        case EVT_FIELD_NAME:
+                            if (firstFieldIndex == -1) {
+                                firstFieldIndex = addresses.size() / 2;
+                            }
+                        case EVT_FIELD_VALUE:
+                            assert firstFieldIndex != -1;
+                            addresses.add(token.getCacheAddress());
+                            break;
+                        case EVT_TIMESTAMP:
+                            assert timestampAddress == 0;
+                            timestampAddress = token.getCacheAddress();
+                            break;
+
+                        default:
+                            throw new RuntimeException("Unrecognised type " + type);
```

Comment:
This un-typed, perhaps not logged correctly and un-tested. In any case we should not be throwing new instance and concatenating strings here.

We need to throw some "bad" line protocol text at the system.

Reply:
This is a failure scenario, bug in our code (not triggered by third party input). I can change it to an assert or an Error if you like